### PR TITLE
Feature/sbachmei/mic 3199 model 0 observers

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*: implementation
- *JIRA issue*: [MIC-3199](https://jira.ihme.washington.edu/browse/MIC-3199)
- *Research reference*: [link](https://vivarium-research.readthedocs.io/en/latest/concept_models/vivarium_us_cvd/concept_model.html#)

Added in vph MortabiliyObserver and DisabilityObserver.

### Verification and Testing
`make_results` now stratifies deaths and ylls. total ylds are in the output
but no stratifications (as expected until model 1)

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium_nih_us_cvd/constants/results.py [hunk 1]
```diff
@@ -28,13 +28,15 @@
...
+# SDB: these don't match the data column names. Why arent the templates centralized?
+# FIXME [MIC-3230]: Update to match template
 TOTAL_POPULATION_COLUMN_TEMPLATE = 'total_population_{POP_STATE}'
-PERSON_TIME_COLUMN_TEMPLATE = 'person_time_in_{YEAR}_among_{SEX}_in_age_group_{AGE_GROUP}'
-DEATH_COLUMN_TEMPLATE = 'death_due_to_{CAUSE_OF_DEATH}_in_{YEAR}_among_{SEX}_in_age_group_{AGE_GROUP}'
-YLLS_COLUMN_TEMPLATE = 'ylls_due_to_{CAUSE_OF_DEATH}_in_{YEAR}_among_{SEX}_in_age_group_{AGE_GROUP}'
-YLDS_COLUMN_TEMPLATE = 'ylds_due_to_{CAUSE_OF_DISABILITY}_in_{YEAR}_among_{SEX}_in_age_group_{AGE_GROUP}'
-STATE_PERSON_TIME_COLUMN_TEMPLATE = '{STATE}_person_time_in_{YEAR}_among_{SEX}_in_age_group_{AGE_GROUP}'
-TRANSITION_COUNT_COLUMN_TEMPLATE = '{TRANSITION}_event_count_in_{YEAR}_among_{SEX}_in_age_group_{AGE_GROUP}'
+PERSON_TIME_COLUMN_TEMPLATE = 'person_time_year_{YEAR}_sex_{SEX}_age_{AGE_GROUP}'
```
Comment (hussain-jafari) [inline src/vivarium_nih_us_cvd/constants/results.py]: I would include these as part of the post processing PR/ticket instead.

File: src/vivarium_nih_us_cvd/constants/results.py [hunk 2]
```diff
@@ -51,10 +53,25 @@
 
 POP_STATES = ('living', 'dead', 'tracked', 'untracked')
 SEXES = ('male', 'female')
-# TODO - add literals for years in the model
-YEARS = ()
-# TODO - add literals for ages in the model
-AGE_GROUPS = ()
+YEARS = tuple(range(2021, 2023))
```
Comment (hussain-jafari) [inline src/vivarium_nih_us_cvd/constants/results.py]: I would include these as part of the post processing PR/ticket instead as well.

File: src/vivarium_nih_us_cvd/model_specifications/nih_us_cvd.yaml [hunk 3]
```diff
@@ -25,11 +30,27 @@ configuration:
...
-        population_size: 50_000
+        population_size: 500
+        # SDB: how would I model only one sex?
+        # SDB: how do I model 3+ but only observe 25+? They seem to match w/ gbd age bins but how to override?
         age_start: 25
         age_end: 125
+    # FIXME [MIC-3230]: update template
+    observers:
+    #     # SDB: where are these specified? Like, 'age' means stratify by age but where are age bins specified?
+    #     # e.g. even though I only include 'male' in results.SEXES, the output.hdf has female
```
Comment (hussain-jafari) [inline src/vivarium_nih_us_cvd/model_specifications/nih_us_cvd.yaml:47]: We can talk about this and the other comments tomorrow.

File: src/vivarium_nih_us_cvd/results_processing/process_results.py [hunk 4]
```diff
@@ -32,22 +32,21 @@ def make_measure_data(data):
     measure_data = MeasureData(
         population=get_population_data(data),
         ylls=get_by_cause_measure_data(data, 'ylls'),
-        ylds=get_by_cause_measure_data(data, 'ylds'),
+        # ylds=get_by_cause_measure_data(data, 'ylds'),
```
Comment (hussain-jafari) [inline src/vivarium_nih_us_cvd/results_processing/process_results.py]: I would also consider this file to be part of the post processing PR.

**Review by hussain-jafari — APPROVED** [review]

File: src/vivarium_nih_us_cvd/components/observers.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+class ResultsStratifier(ResultsStratifier_):
+    """Centralized component for handling results stratification.
+    This should be used as a sub-component for observers.  The observers
+    can then ask this component for population subgroups and labels during
+    results production and have this component manage adjustments to the
+    final column labels for the subgroups.
+    """
+
+    def register_stratifications(self, builder: Builder) -> None:
```
Comment (hussain-jafari) [inline src/vivarium_nih_us_cvd/components/observers.py]: In this case it looks like we can just use ResultsStratifier directly from VPH, but let's talk later today.

**Pull request merged**